catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorIT.java

```diff
@@ -0,0 +1,272 @@
+import static org.codice.ddf.catalog.content.monitor.features.CxfFeatures.CxfFeature.CXF_RT_SECURITY_SAML;
+import static org.codice.ddf.catalog.content.monitor.features.CxfFeatures.CxfFeature.CXF_WS_POLICY;
+import static org.codice.ddf.catalog.content.monitor.features.CxfFeatures.CxfFeature.CXF_WS_SECURITY;
+import static org.codice.ddf.catalog.content.monitor.features.CxfFeatures.cxfFeatures;
+import static org.codice.ddf.catalog.content.monitor.features.KarafSpringFeatures.SpringFeature.SPRING;
+import static org.codice.ddf.catalog.content.monitor.features.KarafSpringFeatures.karafSpringFeatures;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.ops4j.pax.exam.CoreOptions.composite;
+import static org.ops4j.pax.exam.karaf.options.KarafDistributionOption.editConfigurationFilePut;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.inject.Inject;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.Component;
+import org.codice.ddf.catalog.content.monitor.configurators.KeystoreTruststoreConfigurator;
+import org.codice.ddf.catalog.content.monitor.util.BundleInfo;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.stubbing.Answer;
+import org.ops4j.pax.exam.Option;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+
+import com.google.common.io.Files;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.content.operation.CreateStorageRequest;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.security.service.SecurityManager;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+public class ContentDirectoryMonitorIT extends AbstractComponentTest {
+
+    @Inject
+    private CamelContext camelContext;
+
+    @Inject
+    private ContentDirectoryMonitor contentDirectoryMonitor;
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private String directoryPath;
+
+    private CatalogFramework catalogFramework;
+
+    @Override
+    protected Option setupDistribution() {
```

Comment:
✏️  We should keep all the `public` methods at the top of the class instead of having the `protected` and `private` ones. Makes it easier to see what the tests are without having to scroll way down.